pkg/apis/acme/v1alpha2/types_issuer.go

```diff
@@ -317,13 +317,13 @@ type ACMEIssuerDNS01ProviderRoute53 struct {
 // ACMEIssuerDNS01ProviderAzureDNS is a structure containing the
 // configuration for Azure DNS
 type ACMEIssuerDNS01ProviderAzureDNS struct {
-	ClientID string `json:"clientID"`
+	ClientID string `json:"clientID,omitempty"`
 
-	ClientSecret cmmeta.SecretKeySelector `json:"clientSecretSecretRef"`
+	ClientSecret *cmmeta.SecretKeySelector `json:"clientSecretSecretRef,omitempty"`
 
 	SubscriptionID string `json:"subscriptionID"`
 
-	TenantID string `json:"tenantID"`
+	TenantID string `json:"tenantID,omitempty"`
```

Comment:
Can you add `// +optional` above all optional fields, and additionally add a comment to the fields in this structure explaining what they are, and what the behaviour is if left unset?

Reply:
done